app/views/admin/products_v3/_table.html.haml

```diff
@@ -32,15 +32,15 @@
         %th.align-left= t('admin.products_page.columns.category')
         %th.align-left= t('admin.products_page.columns.tax_category')
         %th.align-left= t('admin.products_page.columns.inherits_properties')
-    - products.each do |product, i|
+    - products.each do |product|
       = form.fields_for(product) do |product_form|
         %tbody.relaxed
           %tr
             %td.align-left.header
-              = product_form.hidden_field :id, name: "[products][#{i}][id]" #todo: can we remove #{i} and implicitly pop?
-              = product_form.text_field :name, name: "[products][#{i}][name]", 'aria-label': t('admin.products_page.columns.name')
+              = product_form.hidden_field :id, name: "[products][][id]"
+              = product_form.text_field :name, name: "[products][][name]", 'aria-label': t('admin.products_page.columns.name')
```

Comment:
:exploding_head: How does this work? So maybe everything is put in the first array element with index 0 until the same name is repeated and that's then put in with the next index. That may rely on every input name being present in every row. We should check if this fails:
```
[a][][id]
[a][][name]
[a][][optional_value]
[a][][id]
[a][][name]
# option not available
[a][][id]
[a][][name]
[a][][optional_value]
```

I'm wondering if the optional value of the third row will accidentally be put in the second row. Do you know?

Is there any reason why we can't use the id as index as well? The old sets couldn't do that because they supported creating several new rows without id but you said that we would probably find another UX for creating entries.

Reply:
I agree, this is not easy to read. To answer your question, I'm not sure, Rails probably handles it pretty well but on reflection, it seems unwise to not specify an index. 

The reason we don't use the record ID as index is because we need to maintain the order that the records were initially shown in. For example, if the view is sorted by name and I update a name, the record will suddenly move and it won't be easy to identify what's moved where.

So i will revert this commit next..